src/helpers/client/FormValue.js

```diff
@@ -0,0 +1,89 @@
+/** Handles the core of form values, including:
+ * - Storing both a current and future value
+ * - Tracking when the user manipulates a future value
+ * - Propagating current values to future values if they
+ *     haven't been manipulated by the user
+ * - Making this instance serializable
+ * - Making sure only valid values are saved.
+ * 
+ * All values stored in a FormValue instance should be valid
+ * 
+ * There must be a better way to write that.
+ * @todo Should `FormValue`s contain a reference to their
+ *     client object?
+ * 
+ * @constructor
+ * @param {string} name - the name of the client property
+ * @param {any} [val] - the initial value
+ * @param {function(*, *, string, FormValue): *} [makeValid] - 
+ *     returns a valid value no matter what. Will be given the
+ *     new value, the current value, the string 'current' or
+ *     'future', and the `FormValue` instance.
+ * 
+ * @returns {FormValue} formVal
+ * @todo Finish description of FormValue return value (tbd)
+ */
+class FormValue {
+
+  constructor ( name, current, makeValid ) {
+
+    // Set defaults
+    if ( current === undefined ) { current = null; }
+    this._makeValid = makeValid || function ( newVal ) { return newVal; };
+
+    this.name = name;
+    
+    // Set and 'set' the starting values
+    this._future = current;
+    this._current = current;
+
+  }  // End FormValue.constructor()
+
+  
+  set current ( value ) {
+    this._current = this._makeValid( value, this._current, 'current', this );
+    return this._current;
```

Comment:
`return` values in setters are ignored, I believe (check me though, if you're not sure). Default behavior is that the setter will always return the new value. I would suggest removing this and line 59 for clarity/consistency with most JS setters in the world.

Reply:
`return` values in setters seem to return the value that was sent, not the value that was eventually set.

Reply:
Check the first console log here: https://codepen.io/knod/pen/XzRedx?editors=0010